distribution/docs/src/main/resources/_contents/extending-contents.adoc

```diff
@@ -817,6 +817,51 @@ It is recommended that:
 * The maven project's pom file should be updated to attach each data/configuration file as an artifact (using the `build-helper-maven-plugin`)
 * Add each data/configuration file to the KAR file by using the `<configfile>` tag in the KAR's `features.xml` file
 
+=== Migration API
+
+${branding} currently has an experimental API for making bundles migratable. Interfaces in `platform/migration/platform-migratable-api` are
+used by the system to identify bundles that provide implementations for import (_coming soon_) and export operations.
+
+[NOTE]
+====
+This code is experimental. While this interface is functional and tested, it may change or be removed in a future version of the library.
+====
+
+When an `migration:export` command is processed, `ConfigurationMigrationManager` will do a look-up for all registered OSGi bundles
+that are `Migratable` and call their `export()` method. When import is created, it will first import all data that comes from bundles
+registered as `ConfigurationMigratable` before importing bundles that are `DataMigratable`.
+
+==== Migratable
```

Comment:
Wonder if we should mention this interface here since we don't want anyone to implement it. Also, providing a useful description for it is difficult and may only make things more confusing.